src/inference.py

```diff
@@ -0,0 +1,159 @@
+import argparse
+
+import torch
+import torch.utils.benchmark as benchmark
+
+from src import log
+from src.log import print_requirements
+
+logger = log.logger
+
+ARCHITECTURES = {
+    "resnet50": "resnet50",
+    "convnext": "convnext_base",
+    "vgg16": "vgg16",
+    "efficient_net_v2": "efficientnet_v2_m",
+    "mobilenet_v3": "mobilenet_v3_large",
+    "resnext50": "resnext50_32x4d",
+    "swin": "swin_b",
+    "vit": "vit_b_16",
+    "ssd_vgg16": "ssd300_vgg16",
+    "fasterrcnn_resnet50_v2": "fasterrcnn_resnet50_fpn_v2",
+}
+
+
+def benchmark_inference(
+    stmt: str,
+    setup: str,
+    input: torch.Tensor,
+    n_runs=100,
+    num_threads: int = 1,
+):
+    """
+    Benchmark a model using torch.utils.benchmark.
+
+    When evaluating model speed in MP/s only the video height, width and batch size are taken into
+    account. The number of channels and sequence length are ignored. Speed evaluation measures
+    how fast can we process an arbitrary input video so channels and sequence length don't
```

Comment:
maybe change this docs , video->image, and not mention sequence length anywhere ?